### PR TITLE
Conditional train fixes

### PR DESCRIPTION
**Pull request by sijoerger**
Fixed the conditional training and made it work with compareAll.sh.

### CODE REVIEW

File: batchrunNNApplication.sh [hunk 1]
```diff
@@ -7,10 +7,13 @@ channels=$( echo $2 | tr "," " " )
 cluster=$3
 modus=$4
 tag=$5
+all_eras=$6
 outdir=${era}_${tag}
 source utils/bashFunctionCollection.sh
 #export SCRAM_ARCH="slc6_amd64_gcc700"
 
+echo "All eras is set to $all_eras"
```
Comment (mova) [inline batchrunNNApplication.sh]: should be in a condtion

File: batchrunNNApplication.sh [hunk 2]
```diff
@@ -22,10 +25,10 @@ fi
 
 if [[ $cluster = "etp7" ]]; then
     #### use local resources
-    export sw_src_dir="/portal/ekpbms3/home/${USER}/CMSSW_10_2_14/src"
-    export batch_out="/portal/ekpbms3/home/${USER}/batch-out"
+    export sw_src_dir="/portal/ekpbms2/home/${USER}/CMSSW_10_2_14/src"
```
Comment (mova) [inline batchrunNNApplication.sh:30]: this is is as user specific setting ... Ideally, we should read it from outside the repo... for now, i'd leave it be.

File: compareAll.sh [hunk 3]
```diff
@@ -18,7 +18,7 @@ export PARALLEL=1
 export USE_BATCH_SYSTEM=1
 export cluster=naf7 #lxplus7 # etp7
 export sm_htt_analysis_dir=$( pwd ) ### local sm-htt repo !
-export cmssw_src_local="/portal/ekpbms3/home/${USER}/CMSSW_10_2_14/src" ### local CMSSW !
+export cmssw_src_local="/portal/ekpbms2/home/${USER}/pruning/CMSSW_10_2_14/src" ### local CMSSW !
```
Comment (mova) [inline compareAll.sh:21]: user specific

File: compareAll.sh [hunk 4]
```diff
@@ -49,8 +49,10 @@ IFS=',' read -r -a tags <<< $3
 erasarg=$1
 channelsarg=$2
 tagsarg=$3
+conditional_arg=$4
```
Comment (mova) [inline compareAll.sh]: is this needed?

File: compareAll.sh [hunk 5]
```diff
@@ -358,6 +370,7 @@ function runana() {
 ### generate postfitshape
 function plotPreFitShapes() {
     ensureoutdirs
+    CATEGORIES="stxs_stage1p1"
```
Comment (mova) [inline compareAll.sh]: `export JETFAKES=1 EMBEDDING=1 CATEGORIES="stxs_stage1p1"` ?

File: ml/create_training_dataset.sh [hunk 6]
```diff
@@ -59,7 +64,6 @@ function run_procedure() {
          --output-filename training_dataset.root \
          --tree-path ${CHANNEL}_nominal/ntuple \
          --event-branch event \
-         --training-stxs1p1 \
```
Comment (mova) [inline ml/create_training_dataset.sh:62]: this  should stay the new default

File: utils/setup_cmssw.sh [hunk 7]
```diff
@@ -5,7 +5,7 @@ export VO_CMS_SW_DIR=/cvmfs/cms.cern.ch
 source $VO_CMS_SW_DIR/cmsset_default.sh
 
 THIS_PWD=$PWD
-cd CMSSW_10_2_16_UL/src
+cd CMSSW_10_2_14/src
```
Comment (mova) [inline utils/setup_cmssw.sh]: ?

File: compareAll.sh [hunk 8]
```diff
@@ -49,8 +49,10 @@ IFS=',' read -r -a tags <<< $3
 erasarg=$1
 channelsarg=$2
 tagsarg=$3
+conditional_arg=$4
 
 if [[ $eras == "all" ]]; then
```
Comment (mova) [inline compareAll.sh]: eras->erasarg

**Review by mova — CHANGES_REQUESTED** [review]

File: compareAll.sh [hunk 9]
```diff
@@ -117,9 +119,15 @@ function compenv() {
 
 function genTrainingDS() {
     ensuremldirs
-    for tag in ${tags[@]}; do
-        logandrun ./ml/create_training_dataset.sh ${erasarg} ${channelsarg} ${tag}
-    done
```
Comment (mova) [inline compareAll.sh:122]: `erasarg` should be `all` if you pass ``all`. I dont think this is needed

Comment (sijoerger) [reply compareAll.sh:122]: [[ "" = $( echo ${eras} ) ]] && eras=("2016" "2017" "2018") erasarg="2016,2017,2018"

Ne, diese Zeile macht erasarg zu "2016,2017,2018" wenn man davor "all" hatte.

File: compareAll.sh (hunk 8 above)
Comment (sijoerger) [reply compareAll.sh]: Done

File: compareAll.sh (hunk 4 above)
Comment (sijoerger) [reply compareAll.sh]: Nope, artefakt, vergessen zu löschen

File: batchrunNNApplication.sh (hunk 1 above)
Comment (sijoerger) [reply batchrunNNApplication.sh]: Done

File: compareAll.sh (hunk 5 above)
Comment (sijoerger) [reply compareAll.sh]: true, hatte das drin wegen was davor. Done

File: compareAll.sh (hunk 8 above)
Comment (mova) [reply compareAll.sh]: also, in line 74, all sould be accepted as a valid era.
`if [[ ! "2016 2017 2018" =~ ${era} ]]; then` 
->
`if [[ ! "2016 2017 2018 all" =~ ${era} ]]; then`

File: ml/create_training_dataset.sh (hunk 6 above)
Comment (sijoerger) [reply ml/create_training_dataset.sh:62]: Jupp, sorry

File: utils/setup_cmssw.sh (hunk 7 above)
Comment (sijoerger) [reply utils/setup_cmssw.sh]: Changed

File: compareAll.sh (hunk 9 above)
Comment (mova) [reply compareAll.sh:122]: ah, richtig :/

File: compareAll.sh (hunk 8 above)
Comment (sijoerger) [reply compareAll.sh]: Nicht nötig wegen [[ "" = $( echo ${eras} ) ]] && eras=("2016" "2017" "2018") erasarg="2016,2017,2018"

**Review by mova — APPROVED** [review]

**Pull request merged**